homeassistant/components/aquacell/config_flow.py

```diff
@@ -0,0 +1,130 @@
+"""Config flow for Aquacell integration."""
+
+from __future__ import annotations
+
+import logging
+from typing import Any
+
+from aioaquacell import ApiException, AquacellApi, AuthenticationFailed
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.config_entries import ConfigFlowResult
+from homeassistant.const import CONF_EMAIL, CONF_PASSWORD
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import HomeAssistantError
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+
+from .const import CONF_REFRESH_TOKEN, DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_EMAIL): str,
+        vol.Required(CONF_PASSWORD): str,
+    }
+)
+
+
+async def validate_input(
+    hass: HomeAssistant, email: str, password: str
+) -> dict[str, Any]:
+    """Validate the user input allows us to connect."""
+
+    session = async_get_clientsession(hass)
+    api = AquacellApi(session)
+    refresh_token = None
+    try:
+        refresh_token = await api.authenticate(email, password)
```

Comment:
Is this an oauth2 integration?

Reply:
Unfortunately it's not, it's authenticating through AWS cognito libraries same as the mobile app does. While cognito does also support OAuth2 I haven't managed to get this working.